File: sol/1945/1945.cpp
struct Solution {
    int getLucky(string &s, int k) {
        int res = 0;
        for (char c : s){
            int cn = c - 96;
            res += cn / 10 + cn % 10;
        }
        while (--k > 0){
            int tmp = 0;
            while (res != 0){
                tmp += res % 10;
                res /= 10;
            }
            res = tmp;
        }
        return res;
    }
};

File: sol/1945/1945.py
class Solution:
    def getLucky(self, s: str, k: int) -> int:
        res = 0
        for c in s:
            cn = ord(c) - 96
            q, r = divmod(cn, 10)
            res += q + r
        for _ in range(k - 1):
            temp = 0
            while res != 0:
                res, r = divmod(res, 10)
                temp += r
            res = temp
        return res
